Content/Items/Vitric/Weapons.RecursiveFocus.cs

```diff
@@ -0,0 +1,905 @@
+using Terraria;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Helpers;
+using Terraria.Graphics.Effects;
+using Terraria.Audio;
+using Terraria.DataStructures;
+using Microsoft.Xna.Framework.Audio;
+using ReLogic.Utilities;
+using StarlightRiver.Physics;
+
+namespace StarlightRiver.Content.Items.Vitric
+{
+    public class RecursiveFocus : ModItem
+    {
+        public override string Texture => AssetDirectory.VitricItem + Name; 
+
+        public override void SetStaticDefaults()
+        {
+            Tooltip.SetDefault("Summons an infernal crystal\nThe infernal crystal locks onto enemies, ramping up damage overtime\nPress <right> to cause the crystal to target multiple enemies, at the cost of causing all beams to not ramp up, dealing less damage");
+        }
+
+        public override void SetDefaults()
+        {
+            Item.CloneDefaults(ItemID.QueenSpiderStaff);
+            Item.sentry = true;
+            Item.damage = 14;
+            Item.DamageType = DamageClass.Summon;
+            Item.mana = 15;
+            Item.knockBack = 0f;
+
+            Item.rare = ItemRarityID.Orange;
+            Item.value = Item.sellPrice(gold : 2);
+
+            Item.UseSound = SoundID.Item25;
+            Item.useStyle = ItemUseStyleID.HoldUp;
+
+            Item.shoot = ModContent.ProjectileType<RecursiveFocusProjectile>();
+            Item.shootSpeed = 1f;
+
+            Item.useTime = Item.useAnimation = 35;
+        }
+
+        public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+        {
+            if (player.ownedProjectileCounts[Item.shoot] > 0)
+            {
+                for (int i = 0; i < Main.maxProjectiles; i++)
+                {
+                    Projectile proj = Main.projectile[i];
+                    if (proj.active && proj.type == Item.shoot && proj.owner == player.whoAmI)
+                    {
+                        proj.Kill();
+                    }
+                }
+            }
+            Projectile.NewProjectileDirect(source, player.Center, velocity, type, damage, knockback, player.whoAmI).originalDamage = Item.damage;
+            player.UpdateMaxTurrets();
+            return false;
+        }
+    }
+
+    public class RecursiveFocusProjectile : ModProjectile
+    {
+        public Player owner => Main.player[Projectile.owner];
+
+        public override string Texture => AssetDirectory.VitricItem + Name;
+
+        public int modeSwitchTransitionTimer;
+
+        public int modeSwitchCooldown;
+
+        public int flashTimer;
+
+        public int trailMult = 5;
+
+        public int trailMult2 = 3;
+
+        public int timeSpentOnTarget;
+
+        public int timeSpentOnMultiTargetOne;
+        public int timeSpentOnMultiTargetTwo;
+
+        public Vector2 targetCenter;
+
+        public Vector2 multiTargetCenterOne;
```

Comment:
Some of these fields look redundant, could they not be made into properties getting from the NPC instance?

Also why are there two of each?